lndr-backend/src/Lndr/Db/VerifiedCredits.hs

```diff
@@ -73,15 +73,19 @@ verifyCreditByHash :: Text -> Connection -> IO Int
 verifyCreditByHash hash conn = fromIntegral <$> execute conn "UPDATE settlements SET verified = TRUE WHERE hash = ?" (Only hash)
 
 
+-- Balance is calculated per ucac and only includes bilateral non-settlement
+-- credits and transaction-verified bilateral settlementcredits
 userBalance :: Address -> Address -> Connection -> IO Integer
 userBalance addr ucac conn = do
-    [Only balance] <- query conn "SELECT (SELECT COALESCE(SUM(amount), 0) FROM verified_credits WHERE creditor = ? AND ucac = ?) - (SELECT COALESCE(SUM(amount), 0) FROM verified_credits WHERE debtor = ? AND ucac = ?)" (addr, ucac, addr, ucac) :: IO [Only Scientific]
+    [Only balance] <- query conn "SELECT (SELECT COALESCE(SUM(verified_credits.amount), 0) FROM verified_credits LEFT JOIN settlements USING(hash) WHERE creditor = ? AND ucac = ? AND verified IS DISTINCT FROM FALSE) - (SELECT COALESCE(SUM(verified_credits.amount), 0) FROM verified_credits LEFT JOIN settlements USING (hash) WHERE debtor = ? AND ucac = ? AND verified IS DISTINCT FROM FALSE)" (addr, ucac, addr, ucac) :: IO [Only Scientific]
```

Comment:
In plain English, what is the intention of this query? Is there any way to split it across multiple lines for readability?

Reply:
Is this something that could potentially be more legibly described in Haskell from the results of a more straightforward query?

Reply:
```sql
SELECT (
  SELECT
    COALESCE(
      SUM(verified_credits.amount),
      0
    )
  FROM
    verified_credits
      LEFT JOIN
    settlements
      USING(hash)
  WHERE
    creditor = ?
      AND
    ucac = ?
      AND
    verified IS DISTINCT FROM FALSE
) - (
  SELECT
    COALESCE(
      SUM(verified_credits.amount),
      0
    )
  FROM
    verified_credits
      LEFT JOIN
    settlements
      USING (hash)
  WHERE
    debtor = ?
      AND
    ucac = ?
      AND
    verified IS DISTINCT FROM FALSE
)
```

Reply:
I wanted to stay out of haskell land entirely because this is a frequently-made query and shouldn't require two round trips to the DB. Let me glance at the comments you made below now...